packages/contracts-bedrock/test/Initializable.t.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ERC721Bridge_Initializer } from "test/CommonTest.t.sol";
+import { console } from "forge-std/console.sol";
+import { CrossDomainMessenger } from "src/universal/CrossDomainMessenger.sol";
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+
+/// @title Initializer_Test
+/// @dev Ensures that the `initialize()` function on contracts cannot be called more than
+///      once. This contract inherits from `ERC721Bridge_Initializer` because it is the
+///      deepest contract in the inheritance chain for setting up the system contracts.
```

Comment:
Maybe overkill, but is it worth considering how we can ensure we aren't missing any contracts in this test? It seems we'd currently just verify manually which might be sufficient. A more robust way might be something like:

```solidity
// Define initialization calls for each contract.
struct InitializeableContracts {
  address target;
  bytes initializationCalldata;
}

InitializeableContracts[] contracts;
contracts.push(InitializeableContracts({
    target: address(L1Messenger), 
    initializationCalldata: abi.encodeCall(L1Messenger.initialize, (OptimismPortal(payable(address(0)))))
}));
contracts.push(etc...);

// Read artifacts for all contracts in src/L1.
// --- snip ---

// If it has a method called initialize, increment counter.
uint256 expectedContracts;
// --- snip ---

// Verify no contracts are missing.
assertEq(contracts.length, expectedContracts, "Missing contracts");

// Now execute the initialization calls that we expect to revert.
// --- snip ---
```

Alternatively could even dynamically generate the `InitializeableContracts[] contracts` array from the initialization sigs read from the artifacts

Reply:
This is helpful context, thanks. Given that, I'd support leaving this PR as-is for now, and having an issue to track the modification of this `test_cannotReinitializeL1_succeeds` test to be artifact-based once the change to testing based on deploy scripts is made

Reply:
https://github.com/ethereum-optimism/optimism/issues/7691